### DISCUSSION TITLE
Criar Coluna de porcentagem simples

### DISCUSSION THREAD

**Discussion by jornalistaCarlosOsorio** [sqlitebrowser/sqlitebrowser#2685] [Q&A]
Como faço para criar código para cálculo de porcentagem simples com duas colunas?

SELECIONE
Coluna 01,
Coluna 02,
Porcentagem
DE NOME_DO_ARQUIVO

Realizei algumas amostras, mas não obtive sucesso. Alguém pode me ajudar?

**Comment by chrisjlocke** [comment]
So column 2 is a percentage of column 1?  So if column 1 was 100, and column 2 was 40, you'd want 40% of 100?
Note that SQLite doesn't like dividing two integers - it'll make the result an integer, which you don't want, so you need to use CAST.

```
select cast(column1 as real)/100*column2 as percentage
from tableA
```
As a side-note, ideally column names shouldn't have spaces in them.
Hope this helps!

---
Google translated:

Então a coluna 2 é uma porcentagem da coluna 1? Portanto, se a coluna 1 fosse 100 e a coluna 2 fosse 40, você gostaria de 40% de 100?
Note que o SQLite não gosta de dividir dois inteiros - fará com que o resultado seja um inteiro, o que você não quer, então você precisa usar CAST.

```
select cast(column1 as real)/100*column2 as percentage
from tableA
```
Como uma observação lateral, o ideal é que os nomes das colunas não tenham espaços.
Espero que isto ajude!

**Reply by jornalistaCarlosOsorio** [reply]
O resultado dá um cálculo errado com esse código cast(Total_eleitores as real)/100*Votos as percentual

SELECT
Nome,
Votos, 
Total_eleitores,
cast(Total_eleitores as real)/100*Votos AS Percentual
FROM

coluna votos = 02
coluna Total_eleitores = 380
coluna Percentual = 7.6 (RESULTADO)

O resultado correto = 0.53 (CORRETO)

Infelizmente, não estou conseguindo acertar o código correto. Desde já, agradeço!

**Reply by jornalistaCarlosOsorio** [reply]
Consegui:

cast(100*Votos as real)/Total_eleitores as Percentual

Gostaria de agradecê-lo, pois sem você jamais chegaria ao resultado desejado.

Gratidão,

**Reply by chrisjlocke** [reply]
Yes, sorry - my calculation was based on giving x percent of y, so 2 percent of 380, which is not really what you wanted.
Glad you got there in the end!  

---

Sim, desculpe - meu cálculo foi baseado em fornecer x por cento de y, ou seja, 2 por cento de 380, que não é realmente o que você queria.
Que bom que você chegou lá no final!

**Reply by chrisjlocke** [reply]
You can also add ROUND to change the original 0.52631579 to a better number.
```
round(votos*100/cast(Total_eleitores as real),2)
```

I've used CAST on the second field but it doesn't matter which one, so long as one value is marked as REAL.

---

Você também pode adicionar ROUND para alterar o 0,52631579 original para um número melhor.
```
round(votos*100/cast(Total_eleitores as real),2)
```

Usei CAST no segundo campo, mas não importa qual, desde que pelo menos um valor esteja marcado como REAL.

**Reply by jornalistaCarlosOsorio** [reply]
Legal, mais uma vez obrigado pela sugestão.

**Reply by chrisjlocke** [reply]
Sem problemas. Sempre feliz em ajudar.
